test/Data/Time/PreciseDuration.purs

```diff
@@ -1,27 +1,33 @@
 module Test.Data.Time.PreciseDuration.Spec where
 
+import Data.Time.PreciseDuration
 import Prelude
 
 import Control.Monad.Aff (Aff)
-import Data.BigInt (BigInt, fromString)
+import Data.BigInt as BigInt
+import Data.Decimal (Decimal)
+import Data.Decimal as Decimal
+import Data.Decimal.Extras (isInteger)
 import Data.Maybe (fromJust)
-import Data.Time.PreciseDuration (PreciseDuration(..), day, hour, micro, milli, minute, nano, second, toDays, toHours, toMicroseconds, toMilliseconds, toMinutes, toNanoseconds, toSeconds, toWeeks, week)
+import Data.Traversable (for_)
 import Partial.Unsafe (unsafePartial)
 import Test.Spec (Spec, describe, it)
 import Test.Spec.Assertions (shouldEqual)
 
-unsafeFromString :: String -> BigInt
-unsafeFromString = unsafePartial fromJust <<< fromString
+unsafeFromString :: String -> Decimal
+unsafeFromString = unsafePartial fromJust <<< Decimal.fromString
 
 test
   :: forall r
    . (PreciseDuration -> PreciseDuration)
-  -> (BigInt -> PreciseDuration)
-  -> BigInt
-  -> BigInt
+  -> (Decimal -> PreciseDuration)
+  -> Decimal
+  -> Array Decimal
   -> Aff r Unit
-test fn ctr div input = do
-  fn (Nanoseconds input) `shouldEqual` (ctr $ input * nano / div)
+test fn ctr div inputs = for_ inputs \input -> do
```

Comment:
We could do `test fn ctr div = traverse_ \input -> do` here 🤷‍♂️

Reply:
Imo it's cleaner to have `inputs` before the big `do` block

Reply:
Cleaner than eta reduction of `inputs`? 🙂